src/main/java/org/broadinstitute/hellbender/tools/genomicsdb/GenomicsDBImport.java

```diff
@@ -221,24 +249,169 @@ public void onTraversalStart() {
+     * @return  Feature readers to be imported in the current batch
+     */
+    private Map<String, FeatureReader<VariantContext>> getFeatureReaders(List<String> sampleNames, Map<String, String> sampleNameToFileName, int lowerSampleIndex, int upperSampleIndex) {
+        Map<String, FeatureReader<VariantContext>> sampleToReaderMap = new LinkedHashMap<>();
+
+        for(int i = lowerSampleIndex; i < upperSampleIndex; ++i) {
+            final String sampleName = sampleNames.get(i);
+            assert sampleNameToFileName.containsKey(sampleName);
+
+            final String variantURI = IOUtils.getPath(sampleNameToFileName.get(sampleName)).toAbsolutePath().toUri().toString();
+            final Function<SeekableByteChannel, SeekableByteChannel> cloudWrapper = (cloudPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudPrefetchBuffer, is) : Function.identity());
+            final Function<SeekableByteChannel, SeekableByteChannel> cloudIndexWrapper = (cloudIndexPrefetchBuffer > 0 ? is -> SeekableByteChannelPrefetcher.addPrefetcher(cloudIndexPrefetchBuffer, is) : Function.identity());
+            final AbstractFeatureReader<VariantContext, LineIterator> reader = AbstractFeatureReader.getFeatureReader(variantURI, null, new VCFCodec(), true, cloudWrapper, cloudIndexWrapper);
+
+            assert sampleName.equals(((VCFHeader) reader.getHeader()).getGenotypeSamples().get(0));
+            sampleToReaderMap.put(sampleName, reader);
+        }
+        return sampleToReaderMap;
+    }
+
+    /**
+     * Private method to create a GenomicsDB configuration data structure
+     * instead of sending a long list of parameters to the constructor call
+     *
+     * @return  GenomicsDB import configuration object
+     */
+    private GenomicsDBImportConfiguration.ImportConfiguration createImportConfiguration(
+        final String workspace,
+        final String arrayName,
+        final long variantContextBufferSize,
+        final long segmentSize,
+        final String vidMapJSONFile,
+        final String callsetMapJSONFile,
+        final long lbSampleIndex,
+        final long ubSampleIndex) {
+
+        GenomicsDBImportConfiguration.Partition.Builder pBuilder =
+            GenomicsDBImportConfiguration.Partition.newBuilder();
+
+        // Since, there is one partition for this import, the
+        // begin column partition index is 0
+        GenomicsDBImportConfiguration.Partition partition =
+            pBuilder
+                .setWorkspace(workspace)
+                .setArray(arrayName)
+                .setBegin(0)
+                .build();
+
+        GenomicsDBImportConfiguration.GATK4Integration.Builder gBuilder =
+            GenomicsDBImportConfiguration.GATK4Integration.newBuilder();
+
+        if (!vidMapJSONFile.isEmpty())
+            gBuilder.setOutputVidmapJsonFile(vidMapJSONFile);
+
+        if (!callsetMapJSONFile.isEmpty())
+            gBuilder.setOutputCallsetmapJsonFile(callsetMapJSONFile);
```

Comment:
Yes, they are needed. The first time we call the importer they must be passed. I have added this logic in the next push.